src/routes/ui/mod.rs

```diff
@@ -235,43 +182,14 @@ pub fn settings_profile_update(
 }
 
 #[get("/")]
-pub fn index(
-    flash: Option<FlashMessage>,
-    user: Option<User>,
-    db_conn: db::Connection,
-) -> Result<Page, Error> {
-    let rendered = Page::new().flash(flash).content(html! {
-        header h1 "Rustodon"
-
-        div {
-            @if let Some(user) = user {
-                @let account = user.get_account(&db_conn)?;
-                div {
-                    a href=(account.get_uri()) "your profile"
-                    " | "
-                    form.inline method="post" action="/auth/sign_out" {
-                        input type="hidden" name="stub"
-                        button.link type="submit" name="submit" "sign out."
-                    }
-                }
-
-                form method="post" action="/statuses/create" {
-                    div input name="content_warning" placeholder="content warning" {}
-                    div textarea name="content" {}
-
-                    button type="submit" "post"
-                }
-            } @else {
-                div {
-                    a href="/auth/sign_in" "sign in!"
-                    " | "
-                    a href="/auth/sign_up" "sign up?"
-                }
-            }
-        }
-    });
-
-    Ok(rendered)
+pub fn index(flash: Option<FlashMessage>, account: Option<Account>) -> IndexTemplate<'static> {
+    IndexTemplate {
+        account: account,
+        _parent: BaseTemplate {
+            flash:    flash,
```

Comment:
ugh i wonder if there's a way to get rid of this constant explicit construction of `BaseTemplate`. maybe a macro?

Reply:
Agreed, it was ugly. I’ll give the macro construction a shot.

Reply:
Done, if you wanna check out the format